v2/processor.go

```diff
@@ -115,16 +148,17 @@ func (p *Processor) Start(ctx context.Context) error {
 }
 
 // start starts the processor and blocks until an error occurs or the context is canceled.
-func (p *Processor) start(ctx context.Context) error {
-	log(ctx, "starting processor")
-	messages, err := p.receiver.ReceiveMessages(ctx, p.options.MaxConcurrency, nil)
+func (p *Processor) start(ctx context.Context, index int) error {
+	receiver := p.receivers[index]
+	log(ctx, fmt.Sprintf("starting processor %d", index))
+	messages, err := receiver.ReceiveMessages(ctx, p.options.MaxConcurrency, nil)
 	if err != nil {
-		return err
+		return fmt.Errorf("processor %d failed to receive messages: %w", index, err)
 	}
-	log(ctx, fmt.Sprintf("received %d messages - initial", len(messages)))
+	log(ctx, fmt.Sprintf("processor %d received %d messages - initial", index, len(messages)))
 	processor.Metric.IncMessageReceived(float64(len(messages)))
```

Comment:
wonder if we need to have different metrics for the different receivers

Reply:
the main challenge here is go-shuttle doesn't record namespace or entity name. It will be difficult to label the receivers.

Reply:
you have now "labeled" them with an index.
if you want to identify them more specifically, we'd need to have a way to know the namespace or entity that the receiver is configured with.
or allow the user to give each receiver a name.

that could get rid of that index. you can then store the receivers in a map with the name as a key.
not certain how much better it is :)

Reply:
note: synced offlined and agreed on allowing user to give receiver a name.